test/ha/domainmapping_test.go

```diff
@@ -0,0 +1,220 @@
+	pkgHa "knative.dev/pkg/test/ha"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+	v1test "knative.dev/serving/test/v1"
+)
+
+const (
+	domainmappingDeploymentName = "domain-mapping"
+)
+
+func TestDomainMappingHA(t *testing.T) {
+	if !test.ServingFlags.EnableAlphaFeatures {
+		t.Skip("Alpha features not enabled")
+	}
+
+	ctx, clients := context.Background(), test.Setup(t)
+
+	if err := pkgTest.WaitForDeploymentScale(context.Background(), clients.KubeClient, domainmappingDeploymentName, system.Namespace(), test.ServingFlags.Replicas); err != nil {
+		t.Fatalf("Deployment %s not scaled to %d: %v", domainmappingDeploymentName, test.ServingFlags.Replicas, err)
+	}
+
+	leaders, err := pkgHa.WaitForNewLeaders(context.Background(), t, clients.KubeClient, domainmappingDeploymentName, system.Namespace(), sets.NewString(), test.ServingFlags.Buckets)
+	if err != nil {
+		t.Fatal("Failed to get leader:", err)
+	}
+	t.Log("Got initial leader set:", leaders)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   test.PizzaPlanet1,
+	}
+
+	// Clean up on test failure or interrupt.
+	test.EnsureTearDown(t, clients, &names)
+
+	// Set up initial Service.
+	svc, err := v1test.CreateServiceReady(t, clients, &names)
+	if err != nil {
+		t.Fatalf("Failed to create initial Service %v: %v", names.Service, err)
+	}
+
+	// Using fixed hostnames can lead to conflicts when multiple tests run at
+	// once, so include the svc name to avoid collisions.
+	host := svc.Service.Name + ".example.org"
+	// Set resolvabledomain for custom domain to false by default.
+	resolvableCustomDomain := false
+
+	if test.ServingFlags.CustomDomain != "" {
+		host = svc.Service.Name + "." + test.ServingFlags.CustomDomain
+		resolvableCustomDomain = true
+	}
+	// Point DomainMapping at our service.
+	var dm *v1alpha1.DomainMapping
+	if err := reconciler.RetryTestErrors(func(int) error {
+		dm, err = clients.ServingAlphaClient.DomainMappings.Create(ctx, &v1alpha1.DomainMapping{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      host,
+				Namespace: svc.Service.Namespace,
+			},
+			Spec: v1alpha1.DomainMappingSpec{
+				Ref: duckv1.KReference{
+					Namespace:  svc.Service.Namespace,
+					Name:       svc.Service.Name,
+					APIVersion: "serving.knative.dev/v1",
+					Kind:       "Service",
+				},
+			},
+		}, metav1.CreateOptions{})
+		return err
+	}); err != nil {
+		t.Fatalf("Create(DomainMapping) = %v, expected no error", err)
+	}
+
+	t.Cleanup(func() {
+		clients.ServingAlphaClient.DomainMappings.Delete(ctx, dm.Name, metav1.DeleteOptions{})
+	})
+
+	// Wait for DomainMapping to go Ready.
+	waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
+		state, err := clients.ServingAlphaClient.DomainMappings.Get(context.Background(), dm.Name, metav1.GetOptions{})
+		if err != nil {
+			return true, err
+		}
+
+		return state.IsReady(), nil
```

Comment:
I believe the block
```
if err != nil {
   return true, err
}
return state.IsReady(), nil
```
can be replaced with just
```
return state.IsReady(), err
```
because the `err` is evaluated first in the PollImediate function.